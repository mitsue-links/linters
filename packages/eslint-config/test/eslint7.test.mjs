import path from 'path';
import {lintFile} from './helper.mjs';

describe('ESLint 7.0.0', () => {
    test('7.0.0_ok.mjs', async () => {
        const filePath = '7.0.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('7.0.0_error.mjs', async () => {
        const filePath = '7.0.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 7.1.0', () => {
    test('7.1.0_error.mjs', async () => {
        const filePath = '7.1.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 7.3.0', () => {
    test('7.3.0_ok.mjs', async () => {
        const filePath = '7.3.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });


    test('7.3.0_error.mjs', async () => {
        const filePath = '7.3.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 7.15.0', () => {
    test('7.15.0_ok.mjs', async () => {
        const filePath = '7.15.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('7.15.0_error.mjs', async () => {
        const filePath = '7.15.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});
