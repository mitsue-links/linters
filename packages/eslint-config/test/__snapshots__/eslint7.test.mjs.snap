// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint 7.0.0 7.0.0_error.mjs 1`] = `
[
  {
    "filePath": "7.0.0_error.mjs",
    "messages": [
      {
        "line": 5,
        "ruleId": "default-case-last",
        "severity": 2,
      },
      {
        "line": 16,
        "ruleId": "default-case-last",
        "severity": 2,
      },
      {
        "line": 27,
        "ruleId": "no-useless-backreference",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`ESLint 7.1.0 7.1.0_error.mjs 1`] = `
[
  {
    "filePath": "7.1.0_error.mjs",
    "messages": [
      {
        "line": 4,
        "ruleId": "no-loss-of-precision",
        "severity": 1,
      },
      {
        "line": 10,
        "ruleId": "no-loss-of-precision",
        "severity": 1,
      },
    ],
  },
]
`;

exports[`ESLint 7.3.0 7.3.0_error.mjs 1`] = `
[
  {
    "filePath": "7.3.0_error.mjs",
    "messages": [
      {
        "line": 4,
        "ruleId": "no-promise-executor-return",
        "severity": 2,
      },
      {
        "line": 21,
        "ruleId": "no-unreachable-loop",
        "severity": 1,
      },
    ],
  },
]
`;

exports[`ESLint 7.15.0 7.15.0_error.mjs 1`] = `
[
  {
    "filePath": "7.15.0_error.mjs",
    "messages": [
      {
        "line": 3,
        "ruleId": "no-unsafe-optional-chaining",
        "severity": 2,
      },
    ],
  },
]
`;
